src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1240,7 +1245,8 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
             string schemaName,
             string tableName)
         {
-            DataTable? dataTable = EntitiesDataSet.Tables[tableName];
+            string tableNameWithPrefix = GetTableNameWithPrefix(schemaName, tableName);
```

Comment:
Would this also take care of other object types like views and stored procedures too? It would good to test out similar scenario - views, stored procedures with same names in different schemas.

Reply:
This is a good point. I will test this out, and create an issue to track. For this PR I want to avoid increasing the scope too much since adding in additional coverage may complicate the testing scenarios in ways I can't anticipate right now and don't want to create a blocker. I will add a link to the issue to this PR's description when I have it created.